modules/highgui/src/window_framebuffer.cpp

```diff
@@ -0,0 +1,783 @@
+            newHeight = (int)(windowRect.width / aspect_ratio);
+
+            if (newHeight > windowRect.height)
+            {
+                newWidth = (int)(windowRect.height * aspect_ratio);
+                newHeight = windowRect.height;
+            }
+        }
+
+        if ((newWidth != img.cols) && (newHeight != img.rows))
+        {
+            cv::resize(img, img, cv::Size(newWidth, newHeight), INTER_LINEAR);
+        }
+
+        CV_LOG_INFO(NULL, "UI: Formated image: "
+        << cv::typeToString(img.type()) << " size " << img.size());
+
+        if (backend.getMode() == FB_MODE_EMU)
+        {
+            CV_LOG_WARNING(NULL, "UI: FramebufferWindow::imshow is used in EMU mode");
+            return;
+        }
+
+        if (backend.getFBPointer() == MAP_FAILED)
+        {
+            CV_LOG_ERROR(NULL, "UI: Framebuffer is not mapped");
+            return;
+        }
+
+        // SHOW IMAGE
+        int xOffset = backend.getFBXOffset();
+        int yOffset = backend.getFBYOffset();
+        int fbHeight = backend.getFBHeight();
+        int fbWidth = backend.getFBWidth();
+        int lineLength = backend.getFBLineLength();
+
+        int img_start_x;
+        int img_start_y;
+        int img_end_x;
+        int img_end_y;
+        int fb_start_x;
+        int fb_start_y;
+
+        if (windowRect.y - yOffset < 0)
+        {
+            img_start_y = - (windowRect.y - yOffset);
+        }
+        else
+        {
+            img_start_y = 0;
+        }
+        if (windowRect.x - xOffset < 0)
+        {
+            img_start_x = - (windowRect.x - xOffset);
+        }
+        else
+        {
+            img_start_x = 0;
+        }
+
+        if (windowRect.y + yOffset + img.rows > fbHeight)
+        {
+            img_end_y = fbHeight - windowRect.y - yOffset;
+        }
+        else
+        {
+            img_end_y = img.rows;
+        }
+        if (windowRect.x + xOffset + img.cols > fbWidth)
+        {
+            img_end_x = fbWidth - windowRect.x - xOffset;
+        }
+        else
+        {
+            img_end_x = img.cols;
+        }
+
+        if (windowRect.y + yOffset >= 0)
+        {
+            fb_start_y = windowRect.y + yOffset;
+        }
+        else
+        {
+            fb_start_y = 0;
+        }
+        if (windowRect.x + xOffset >= 0)
+        {
+            fb_start_x = windowRect.x + xOffset;
+        }
+        else
+        {
+            fb_start_x = 0;
+        }
+
+        for (int y = img_start_y; y < img_end_y; y++)
+        {
+            std::memcpy(backend.getFBPointer() +
+            (fb_start_y + y - img_start_y) * lineLength + fb_start_x * cntChannel,
+            img.ptr<unsigned char>(y) + img_start_x * cntChannel,
+            (img_end_x - img_start_x) * cntChannel);
+        }
+    }
+
+    double FramebufferWindow::getProperty(int prop) const
+    {
+        CV_LOG_INFO(NULL, "UI: FramebufferWindow::getProperty(int prop: " << prop << ")");
```

Comment:
LOG info is redundant here. The following warning will be shown anyway.

Reply:
fixed